irohad/network/impl/block_loader_service.cpp

```diff
@@ -35,9 +34,9 @@ grpc::Status BlockLoaderService::retrieveBlocks(
     const proto::BlocksRequest *request,
     ::grpc::ServerWriter<::iroha::protocol::Block> *writer) {
   storage_->getBlocksFrom(request->height())
-      .map([this](auto block) {
-        return factory_.serialize(
-            *std::unique_ptr<iroha::model::Block>(block->makeOldModel()));
+      .map([](auto block) {
+        return std::static_pointer_cast<shared_model::proto::Block>(block)
```

Comment:
Maybe better to add an assertion, for cast checking?

Reply:
How  about `dynamic_cast`?